app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -54,31 +94,57 @@ public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
   protected void onLayout(boolean changed, int l, int t, int r, int b) {
     super.onLayout(changed, l, t, r, b);
 
-    indentationLinePath.reset();
     for (int i = 0; i < indentationDepth; i++) {
       float lineStartX = spacePerDepthPx * (i + 1) + indentationLinePaint.getStrokeWidth();
-      indentationLinePath.moveTo(lineStartX, 0);
-      indentationLinePath.lineTo(lineStartX, getHeight());
+
+      this.trees.get(i).path.moveTo(lineStartX, 0);
+      this.trees.get(i).path.lineTo(lineStartX, getHeight());
     }
   }
 
   @Override
   public void draw(Canvas canvas) {
     super.draw(canvas);
-    canvas.drawPath(indentationLinePath, indentationLinePaint);
+
+    for (ColoredTree tree : trees) {
+      indentationLinePaint.setColor(tree.color);
+      canvas.drawPath(tree.path, indentationLinePaint);
+    }
   }
 
   public void setIndentationDepth(@IntRange(from = 0, to = 1) int depth) {
     indentationDepth = depth;
 
+    this.setupDepthLines(depth);
+
     int indentationSpacing = (int) (indentationDepth * spacePerDepthPx + indentationLinePaint.getStrokeWidth());
     setPaddingRelative(originalPaddingStart + indentationSpacing, getPaddingTop(), getPaddingEnd(), getPaddingBottom());
 
     invalidate();
   }
 
+  private void setupDepthLines(int depth) {
+    trees = new ArrayList<>();
+
+    for (int i = 0; i < depth; i++) {
+      Path linePath = new Path();
+      trees.add(new ColoredTree((this.coloredDepthPreference.get() ? Color.parseColor(INDENTATION_COLORS[(i > INDENTATION_COLORS.length ? INDENTATION_COLORS.length : i)]) : defaultIndentationLineColor), linePath));
```

Comment:
I would suggest splitting this into few lines. Right now there is going a bit too much in a single line which makes it harder to read.

Reply:
The logic for handling out of bounds situation seems incorrect to me. The index shouldn't be equal to the length of array too since they start from 0.

My suggestion:

```java
int colorIndex = i % INDENTATION_COLORS.length
```

This will make the colors loop once they reach end of array.

Reply:
Interesting line! There are more colors than possible indents but one never knows.